src/kit.ts

```diff
@@ -467,10 +467,20 @@ function kitHostTargetArch(hostArch: string, targetArch?: string, amd64Alias: bo
     }
   }
 
-  if (targetArch) {
-    return hostArch === targetArch ? hostArch : `${hostArch}_${targetArch}`;
+  if (!targetArch) {
+    targetArch = hostArch;
   }
-  return hostArch;
+
+  // CMake preferred generator platform requires 'win32', while vcvars are still using 'x86'.
+  // This function is called only for VS generators, so it is safe to overwrite
+  // targetArch with the vcvars naming.
+  // In case of any future new mismatches, use the vsArchFromGeneratorPlatform table
+  // instead of hard coding for win32 and x86.
+  // Currently, there is no need of a similar overwrite operation on hostArch,
+  // because CMake host target does not have the same name mismatch with VS.
+  targetArch = vsArchFromGeneratorPlatform[targetArch] || targetArch;
```

Comment:
Should you also add x64 -> amd64 into `vsArchFromGeneratorPlatform` and remove lines 465-467?

Reply:
Oh, maybe not.  That transformation is conditional on `amd64Alias`.